lib/events.js

```diff
@@ -146,24 +141,41 @@ MapboxEventManager.prototype = {
    * @private
    * @param {String} event the name of the event to send to the events service. Valid options are 'search.start', 'search.select', 'search.feedback'.
    * @param {Object} geocoder a mapbox-gl-geocoder instance
+   * @param {Object} eventArgs Additional arguments needed for certain event types
+   * @param {Object} eventArgs.key The key pressed by the user
+   * @param {Object} eventArgs.selectedFeature GeoJSON Feature selected by the user
    * @returns {Object} an event payload
    */
-  getEventPayload: function (event, geocoder) {
+  getEventPayload: function (event, geocoder, eventArgs = {}) {
+    // Make sure required arguments are present for certain event types
+    if (
+      (event === 'search.select' && !eventArgs.selectedFeature) ||
+      (event === 'search.keystroke' && !eventArgs.key)
+    ) {
+      return null;
+    }
+
     // Handle proximity, whether null, lat/lng coordinate object, or 'ip'
     var proximity;
     if (!geocoder.options.proximity) {
       proximity = null;
     } else if (typeof geocoder.options.proximity === 'object') {
       proximity = [geocoder.options.proximity.longitude, geocoder.options.proximity.latitude];
     } else if (geocoder.options.proximity === 'ip') {
-      proximity = [999,999];  // Alias for 'ip' in event logs
+      var ipProximityHeader = geocoder._headers ? geocoder._headers['ip-proximity'] : null;
+      if (ipProximityHeader) {
```

Comment:
nit; Do you think it worth to check for `typeof ipProximityHeader === 'string'` to be sure function `split` is availble.

Reply:
Good idea